packages/sdk/src/actions/executeQuote.ts

```diff
@@ -281,6 +285,109 @@ export async function executeQuote(params: ExecuteQuoteParams): Promise<ExecuteQ
 
         onProgressHandler(currentTransactionProgress);
 
+        // Check if wallet supports atomic transactions
+        const capabilities = await walletClient.getCapabilities({
+          account,
+          chainId: deposit.originChainId,
+        });
+
+        if (
+          capabilities?.atomic?.status === "supported" ||
+          capabilities?.atomic?.status === "ready"
+        ) {
+          try {
+            logger?.debug(
+              "Wallet supports atomic sendCalls, triggering atomic flow",
+            );
+            // Simulate both approval and deposit calls
+            const { calls } = await prepareAtomicTx({
+              walletClient,
+              // publicClient: originClient, #TODO: uncomment this when more public RPCs support eth_simulateV1
```

Comment:
Somewhat related to my comment above. But wdyt of making this whole flow opt-in for now? Consumers of the library should then make sure that if they want their users to use this feature, they use RPC providers that rupport `eth_simulateV1`. We could then remove the inline comments and just throw with a respective error message if the provider doesn't support the call.

Reply:
Makes sense, will update!